README.md

```diff
@@ -1,106 +1,40 @@
-          sub-projects you'd like to additionally build. Can include any of: clang,
-          clang-tools-extra, libcxx, libcxxabi, libunwind, lldb, compiler-rt, lld,
-          polly, or debuginfo-tests.
-
-          For example, to build LLVM, Clang, libcxx, and libcxxabi, use
-          ``-DLLVM_ENABLE_PROJECTS="clang;libcxx;libcxxabi"``.
-
-        * ``-DCMAKE_INSTALL_PREFIX=directory`` --- Specify for *directory* the full
-          path name of where you want the LLVM tools and libraries to be installed
-          (default ``/usr/local``).
-
-        * ``-DCMAKE_BUILD_TYPE=type`` --- Valid options for *type* are Debug,
-          Release, RelWithDebInfo, and MinSizeRel. Default is Debug.
-
-        * ``-DLLVM_ENABLE_ASSERTIONS=On`` --- Compile with assertion checks enabled
-          (default is Yes for Debug builds, No for all other build types).
-
-      * ``cmake --build build [-- [options] <target>]`` or your build system specified above
-        directly.
-
-        * The default target (i.e. ``ninja`` or ``make``) will build all of LLVM.
-
-        * The ``check-all`` target (i.e. ``ninja check-all``) will run the
-          regression tests to ensure everything is in working order.
-
-        * CMake will generate targets for each tool and library, and most
-          LLVM sub-projects generate their own ``check-<project>`` target.
-
-        * Running a serial build will be **slow**.  To improve speed, try running a
-          parallel build.  That's done by default in Ninja; for ``make``, use the option
-          ``-j NNN``, where ``NNN`` is the number of parallel jobs, e.g. the number of
-          CPUs you have.
-
-      * For more information see [CMake](https://llvm.org/docs/CMake.html)
-
-Consult the
-[Getting Started with LLVM](https://llvm.org/docs/GettingStarted.html#getting-started-with-llvm)
-page for detailed information on configuring and compiling LLVM. You can visit
-[Directory Layout](https://llvm.org/docs/GettingStarted.html#directory-layout)
-to learn about the layout of the source code tree.
+## Build instructions
+
+### Requirements 
+- Working C and C++ toolchains(compiler, linker)
+- cmake
+- make or ninja
+
+### 0. Clone Polygeist
+```sh
+git clone --recursive https://github.com/wsmoses/Polygeist.git $SOURCE_DIR
+cd $SOURCE_DIR
+```
+
+### 1. Install LLVM, MLIR, and Clang
+```sh
+mkdir $SOURCE_DIR/llvm-project/build
+cd $SOURCE_DIR/llvm-project/build
+cmake -G Ninja ../llvm \
+  -DLLVM_ENABLE_PROJECTS="mlir;clang" \
+  -DLLVM_TARGETS_TO_BUILD="X86" \
+  -DLLVM_ENABLE_ASSERTIONS=ON \
+  -DCMAKE_BUILD_TYPE=DEBUG
+ninja
+ninja check-mlir
+```
+
+### 2. Install Polygeist
+```sh
+mkdir $SOURCE_DIR/build
+cd $SOURCE_DIR/build
+cmake -G Ninja .. \
+  -DMLIR_DIR=$PWD/../llvm-project/build/lib/cmake/mlir \
+  -DCLANG_DIR=$PWD/../llvm-project/build/lib/cmake/clang \
+  -DLLVM_ENABLE_ASSERTIONS=ON \
+  -DCMAKE_BUILD_TYPE=DEBUG
+ninja
+ninja check-mlir-clang
```

Comment:
Could you add an example?

If possible, a couple of other things to add:
* You will almost certainly want to add the resource directory of the clang built here to the command line `-resource-dir ~/git/Polygeist/mlir-build/lib/clang/14.0.0/`
* Defines, libraries, includes, etc are supported but require space separation.

Perhaps a full example with a 2 line c code?

Reply:
I'd suggest to add a more comprehensive command line tool reference for this? Something like the [command line guide of clang](https://github.com/llvm/llvm-project/blob/c2574e63ff71c1d3caea48cb6200c2422bd8f33d/clang/docs/CommandGuide/clang.rst).

Reply:
Fair enough -- and for that (and other things) we should actually put content in the placeholder website branch.

Reply:
Makes sense to me!